src/net/toxme.cpp

```diff
@@ -149,17 +185,80 @@ bool Toxme::createAddress(ToxId id, QString address,
                           "\"bio\":\""+bio+"\","
                           "\"timestamp\":"+QString().setNum(time(0))+"}"};
 
-    QByteArray response = makeJsonRequest(prepareEncryptedJson(1,payload));
+    QString pubkeyUrl = pubkeyUrls.value(server);
+    QString apiUrl = apiUrls.value(server);
+    QByteArray response = makeJsonRequest(apiUrl, prepareEncryptedJson(pubkeyUrl, 1, payload));
 
-    return (extractError(response) == 0);
+    int errorCode = extractError(response);
+    if (errorCode != 0 && errorCode != INT_MIN)
+        return QString::number(extractError(response));
+    else
+        return getPass(response);
+}
+
+QString Toxme::getPass(QString json) {
+    static const QByteArray pattern{"password\":\""};
+
+    json = json.remove(' ');
+    const int start = json.indexOf(pattern);
+    if (start == -1)
+        return QString();
+
+    json = json.mid(start+pattern.size());
+
+    int end = json.indexOf("\"", start);
+    if (end == -1)
+        return QString();
+
+    json.truncate(end);
+
+    return json;
 }
 
-bool Toxme::deleteAddress(ToxId id)
+bool Toxme::deleteAddress(QString server, ToxId id)
 {
     const QString payload{"{\"public_key\":\""+id.toString().left(64)+"\","
                           "\"timestamp\":"+QString().setNum(time(0))+"}"};
 
-    QByteArray response = makeJsonRequest(prepareEncryptedJson(2,payload));
+    QString pubkeyUrl = pubkeyUrls.value(server);
+    QString apiUrl = apiUrls.value(server);
+    QByteArray response = makeJsonRequest(apiUrl, prepareEncryptedJson(pubkeyUrl, 2, payload));
 
     return (extractError(response) == 0);
 }
+
+QString Toxme::getErrorMessage(int errorCode)
+{
+    switch (errorCode) {
+    case -1:
+        return "You must send POST requests to /api";
+    case -2:
+        return "Please try again using a HTTPS connection";
+    case -3:
+        return "I was unable to read your encrypted payload";
+    case -4:
+        return "You're making too many requests. Wait an hour and try again";
+    case -25:
+        return "This name is already in use";
+    case -26:
+        return "This Tox ID is already registered under another name";
+    case -27:
+        return "Please don't use a space in your name";
+    case -28:
+        return "Password incorrect";
+    case -29:
+        return "You can't use this name";
+    case -30:
+        return "Name not found";
+    case -31:
+        return "Tox ID not sent";
+    case -41:
+        return "Lookup failed because the other server replied with invalid data";
+    case -42:
+        return "That user does not exist";
+    case -43:
+        return "Internal lookup error. Please file a bug";
+    default:
+        return "Unknow error" + errorCode;
+    }
```

Comment:
Yes, I google it :)
What i need to use tr?
Now I have error: 'tr' was not declared in this scope